src/adapters/nats.js

```diff
@@ -0,0 +1,436 @@
+	}
+
+	/**
+	 * Disconnect from adapter
+	 */
+	async disconnect() {
+		this.stopping = true;
+
+		try {
+			if (this.connection) {
+				this.logger.info("Closing NATS JetStream connection...");
+				await this.connection.drain();
+				await this.connection.close();
+
+				this.logger.info("NATS JetStream connection closed.");
+			}
+		} catch (error) {
+			this.logger.error("Error while closing NATS JetStream connection.", error);
+		}
+	}
+
+	/**
+	 * Subscribe to a channel with a handler.
+	 *
+	 * @param {Channel & NATSOpts} chan
+	 */
+	async subscribe(chan) {
+		this.logger.info(`NATS --- ${chan.id} --- in <=`);
+
+		if (chan.maxInFlight == null) chan.maxInFlight = this.opts.maxInFlight;
+		if (chan.maxRetries == null) chan.maxRetries = this.opts.maxRetries;
+
+		chan.deadLettering = _.defaultsDeep({}, chan.deadLettering, this.opts.deadLettering);
+		if (chan.deadLettering.enabled) {
+			chan.deadLettering.queueName = this.addPrefixTopic(chan.deadLettering.queueName);
+		}
+
+		// 1. Create stream
+		// NATS Stream name does not support: spaces, tabs, period (.), greater than (>) or asterisk (*) are prohibited.
+		// More info: https://docs.nats.io/jetstream/administration/naming
+		const streamName = chan.name.split(".").join("_");
+		this.createStream(streamName, [chan.name], chan.nats ? chan.nats.streamConfig : {});
+
+		if (chan.deadLettering && chan.deadLettering.enabled) {
+			const deadLetteringStreamName = chan.deadLettering.queueName.split(".").join("_");
+			this.createStream(
+				deadLetteringStreamName,
+				[chan.deadLettering.queueName],
+				chan.nats ? chan.nats.streamConfig : {}
+			);
+		}
+
+		// 2. Configure NATS consumer
+		this.initChannelActiveMessages(chan.id);
+
+		/** @type {ConsumerOptsBuilder} 
+		const consumerOpts = NATS.consumerOpts();
+		consumerOpts.queue(streamName);
+		// consumerOpts.durable(streamName);
+		consumerOpts.durable(chan.group.split(".").join("_"));
+		consumerOpts.deliverTo(chan.id);
+		consumerOpts.manualAck();
+		consumerOpts.deliverNew();
+		consumerOpts.ackExplicit();
+		consumerOpts.maxAckPending(chan.maxInFlight);
+		// consumerOpts.maxDeliver(chan.maxRetries);
+		// Register the actual handler
+		consumerOpts.callback(this.createConsumerHandler(chan));
+		*/
+
+		/** @type {ConsumerOpts} More info: https://docs.nats.io/jetstream/concepts/consumers */
+		const consumerOpts = _.defaultsDeep(
+			{},
+			chan.nats ? chan.nats.consumerOpts : {},
+			this.opts.nats.consumerOpts
+		);
+
+		consumerOpts.queue = streamName;
+		consumerOpts.config.deliver_group = streamName;
+		// NATS Stream name does not support: spaces, tabs, period (.), greater than (>) or asterisk (*) are prohibited.
+		// More info: https://docs.nats.io/jetstream/administration/naming
+		consumerOpts.config.durable_name = chan.group.split(".").join("_");
+		consumerOpts.config.deliver_subject = chan.id;
+		consumerOpts.callbackFn = this.createConsumerHandler(chan);
```

Comment:
Is it missing the `consumerOpts.maxAckPending(chan.maxInFlight);`?

Reply:
Yes, you're right